### DISCUSSION TITLE
USB RNDIS support for U5A5

### DISCUSSION THREAD

**Discussion by darshanmirajkar-eaton** [zephyrproject-rtos/zephyr#64895] [Q&A]
STM32U575 having Full speed USB and RNDIS enabled in zephyr, For U5A5 Supports USB High speed wrt to below attached image form datasheet.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/118359137/bdbabc2b-2647-4af6-a520-42bcb61fb759)

It would be good to have support for USB and RNDIS enable for U5A5. 

Or Someone suggest how to enable this.

**Comment by erwango** [comment]
This is already supported using following configurations:

# USB Device Settings
CONFIG_USB_DEVICE_STACK=y

# Select USB Configurations
CONFIG_USB_DEVICE_NETWORK_ECM=y
CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT=n

CONFIG_USB_DEVICE_NETWORK_ECM=n
CONFIG_USB_DEVICE_NETWORK_RNDIS=y

**Reply by darshanmirajkar-eaton** [reply]
This works for U575 not working for U5A5

**Reply by erwango** [reply]
Please provide more information when raising issues. We have quite little time to play guessing

**Reply by darshanmirajkar-eaton** [reply]
/ {
	soc {
		usbotg_hs: otgfs@42040000 {
			compatible = "st,stm32-otghs";
			reg = <0x42040000 0x400>;
			interrupts = <73 0>;
			interrupt-names = "otgfs";
			num-bidir-endpoints = <6>;
			ram-size = <1280>;
			maximum-speed = "full-speed";
			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00004000>;
			phys = <&otgfs_phy>;
			status = "disabled";
		};
	};

	otgfs_phy: otgfs_phy {
		compatible = "usb-nop-xceiv";
		#phy-cells = <0>;
	};
};

tried adding this and enabling CONFIG_USB_DC_HAS_HS_SUPPORT=y

**Reply by erwango** [reply]
What is your board configuration ? How does it differ from test made on U575?
What are the boards references you are using ?
I guess this is not the same board, is USB working on U5A5 board?

Please, put you in my shoes and see what information would be useful to help.
Basic assumption is that it should work in the same way on both targets.

**Reply by darshanmirajkar-eaton** [reply]
Only observation is U575 has only Full seed support and U5A5 is High Speed Support

**Reply by darshansm** [reply]
@FRASTM can you suggest anything on this??

**Reply by erwango** [reply]
@Desvauxm-st, can you have a look ?

**Reply by darshanmirajkar-eaton** [reply]
@Desvauxm-st do you have any update on this??

**Reply by darshanmirajkar-eaton** [reply]
@erwango @Desvauxm-st  Any update on this.

**Reply by erwango** [reply]
@darshanmirajkar-eaton This is under investigation. If this point is critical for you, please contact your local ST support.

**Reply by erwango** [reply]
@darshanmirajkar-eaton We're facing an issue in U5 HAL. It appears this configuration isn't working on STM32Cube fw neither.
To speed up things, I'd suggest testing using CubeMX (**using USB device API**). You will face the same issue. Then you can raise an issue in https://github.com/STMicroelectronics/STM32CubeU5

**Comment by Desvauxm-st** [comment]
@darshanmirajkar-eaton : see the word document sent to internal ST buzzila 
[Buzzila for CUBE NUCLEO-U5A5.docx](https://github.com/zephyrproject-rtos/zephyr/files/13471585/Buzzila.for.CUBE.NUCLEO-U5A5.docx)

**Comment by darshanmirajkar-eaton** [comment]
I tried https://github.com/STMicroelectronics/STM32CubeU5/tree/v1.3.0/Projects/NUCLEO-U5A5ZJ-Q/Applications/USBX/Ux_Device_CDC_ACM with this also...USB is not initializing

**Comment by darshanmirajkar-eaton** [comment]
Able to enable RNDIS in U5A5 that in v3.2.0. Suggesting to use PLL1P/2 or HSE as clock source for USB_HS block. For further check this
[https://github.com/STMicroelectronics/STM32CubeU5/issues/34](https://github.com/STMicroelectronics/STM32CubeU5/issues/34)

**Comment by FRASTM** [comment]
@darshanmirajkar-eaton  As we now have a PR for USB OTG HS on the stm32U59x/5Ax with https://github.com/zephyrproject-rtos/zephyr/pull/76200, maybe you could re-test on your side. 
What is the application you are running to have RNDIS enabled ?
